portable/NetworkInterface/xilinx_ultrascale/NetworkInterface.c

```diff
@@ -165,87 +165,150 @@ static u32 ulPHYIndex;
+                 */
+                break;
+            }
 
-        setup_isr( &xEMACpsif );
-        init_dma( &xEMACpsif );
-        start_emacps( &xEMACpsif );
+            XEmacPs_SetOperatingSpeed( pxEMAC_PS, ulLinkSpeed );
 
-        prvGMACWaitLS( xWaitLinkDelay );
+            /* Setting the operating speed of the MAC needs a delay. */
+            vTaskDelay( pdMS_TO_TICKS( 25UL ) );
 
-        /* The deferred interrupt handler task is created at the highest
-         * possible priority to ensure the interrupt handler can return directly
-         * to it.  The task's handle is stored in xEMACTaskHandle so interrupts can
-         * notify the task when there is something to process. */
-        xTaskCreate( prvEMACHandlerTask, "EMAC", configEMAC_TASK_STACK_SIZE, NULL, niEMAC_HANDLER_TASK_PRIORITY, &xEMACTaskHandle );
-    }
-    else
-    {
-        /* Initialisation was already performed, just wait for the link. */
-        prvGMACWaitLS( xWaitRelinkDelay );
-    }
+            ulDMAReg = XEmacPs_ReadReg( pxEMAC_PS->Config.BaseAddress, XEMACPS_DMACR_OFFSET );
+            /* Enable 16-bytes AHB bursts */
+            ulDMAReg = ulDMAReg | XEMACPS_DMACR_INCR16_AHB_BURST;
+
+            /* DISC_WHEN_NO_AHB: when set, the GEM DMA will automatically discard receive
+             * packets from the receiver packet buffer memory when no AHB resource is available. */
+            XEmacPs_WriteReg( pxEMAC_PS->Config.BaseAddress, XEMACPS_DMACR_OFFSET,
+                              ulDMAReg /*| XEMACPS_DMACR_DISC_WHEN_NO_AHB_MASK*/ );
 
-    /* Only return pdTRUE when the Link Status of the PHY is high, otherwise the
-     * DHCP process and all other communication will fail. */
-    xLinkStatus = xGetPhyLinkStatus();
+            setup_isr( &xEMACpsif );
+            init_dma( &xEMACpsif );
+            start_emacps( &xEMACpsif );
+            eEMACState = xEMAC_WaitPHY;
 
-    return( xLinkStatus != pdFALSE );
+        /* Fall through. */
+
+        case xEMAC_WaitPHY:
+            prvGMACWaitLS( xWaitLinkDelay );
+
+            if( xGetPhyLinkStatus() != pdFALSE )
+            {
+                if( xEMACTaskHandle == NULL )
+                {
+                    /* The deferred interrupt handler task is created at the highest
+                     * possible priority to ensure the interrupt handler can return directly
+                     * to it.  The task's handle is stored in xEMACTaskHandle so interrupts can
+                     * notify the task when there is something to process. */
+                    xTaskCreate( prvEMACHandlerTask, "EMAC", configEMAC_TASK_STACK_SIZE, NULL, niEMAC_HANDLER_TASK_PRIORITY, &xEMACTaskHandle );
+                }
+
+                /* Only return pdTRUE when the Link Status of the PHY is high, otherwise the
+                 * DHCP process and all other communication will fail. */
+
+                xReturn = pdPASS;
+            }
+
+            eEMACState = xEMAC_Ready;
+
+            break;
```

Comment:
I did not understand the purpose of this break. Why not fall through?

Reply:
Hello @AniruddhaKanhere , thank you for these comments. I made a few changes, and also I attach a file called [`xilinx_us_driver.zip`](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/files/8338995/xilinx_us_driver.zip)

It shows symbolically how initialisation takes place, leaving out the details.

Also I ran my Xilinx UltraScale testing project:
- A startup while LAN connected: goes to `xEMAC_Ready`
- A startup where LAN is connected after 30 seconds: stays in `xEMAC_SetupPHY`
- Disconnect/connect the LAN while running

All tests went as expected.